### PR TITLE
8 patch article

### PR DESCRIPTION
**Pull request by LukeHarrisonDev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+            const newVote = {inc_votes: 1}
+            return request(app)
+            .patch("/api/articles/1")
+            .send(newVote)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
```
Comment (nc-mark-t) [inline __tests__/app.test.js:151]: You know the article ID, actually, you know the entire article. Where we can we should be specific. We have little use for `any`s here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String)
+                })
+            })
+        })
+        test("PATCH 200: Responds with an article objects votes incremented by 1", () => {
```
Comment (nc-mark-t) [inline __tests__/app.test.js:159]: This could be covered as part of the previous test

File: __tests__/app.test.js [hunk 3]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+            const newVote = {inc_votes: 1}
+            return request(app)
+            .patch("/api/articles/1")
+            .send(newVote)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.article.votes).toBe(101)
+            })
+        })
+        test("PATCH 200: Responds with an article objects votes incremented by more than 1", () => {
```
Comment (nc-mark-t) [inline __tests__/app.test.js:169]: This is potentially unnecessary

File: __tests__/app.test.js [hunk 4]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+            const newVote = {inc_votes: 102}
+            return request(app)
+            .patch("/api/articles/1")
+            .send(newVote)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.article.votes).toBe(202)
+            })
+        })
+        test("PATCH 200: Responds with an article objects votes decremented by 1", () => {
```
Comment (nc-mark-t) [inline __tests__/app.test.js:179]: I like that you checked you can decrement and increment

File: __tests__/app.test.js [hunk 5]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+            const newVote = {inc_votes: -55}
+            return request(app)
+            .patch("/api/articles/1")
+            .send(newVote)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.article.votes).toBe(45)
+            })
+        })
+        test("PATCH 200: Responds with an article objects votes incremented by more than 1", () => {
```
Comment (nc-mark-t) [inline __tests__/app.test.js:199]: This is still a decrement, I also don't think a lot of these tests are testing different functionality

File: __tests__/app.test.js [hunk 6]
```diff
@@ -137,6 +137,96 @@ describe("/api/articles/:article_id", () => {
...
+        test("PATCH 400: Responds with a 'Bad request' when sending an empty object on the request body", () => {
+            return request(app)
+            .patch("/api/articles/1")
+            .send({})
+            .expect(400)
+            .then(({body}) => {
+                expect(body).toEqual({message: "Bad request"})
+            })
+        })
+    })
```
Comment (nc-mark-t) [inline __tests__/app.test.js:229]: What happens if you try to patch a resource that doesn't exist?

File: models/articles.models.js [hunk 7]
```diff
@@ -27,4 +27,31 @@ function fetchArticles() {
...
+    if (typeof votes !== 'number') {
+        return Promise.reject({status:400, message: "Bad request"})
+    }
+    return db.query(
+        `SELECT votes
+        FROM articles
+        WHERE article_id = $1;`, [id])
+    .then(({rows}) => {
+        let currentVotes = rows[0].votes
+        let totalVotes = currentVotes += votes
```
Comment (nc-mark-t) [inline models/articles.models.js:40]: You could do this in the SQL with `SET votes = votes + $1` this would save you a database request

**Review by nc-mark-t — COMMENTED** [review]
Good ticket 👍  I think some of your testing may be overkill but other than one test description they are not incorrect.

File: __tests__/app.test.js (hunk 1 above)
Comment (LukeHarrisonDev) [reply __tests__/app.test.js:151]: Makes sense, cheers

Comment (LukeHarrisonDev) [reply __tests__/app.test.js:151]: Cool, makes sense

File: __tests__/app.test.js (hunk 2 above)
Comment (LukeHarrisonDev) [reply __tests__/app.test.js:159]: No problem

File: __tests__/app.test.js (hunk 3 above)
Comment (LukeHarrisonDev) [reply __tests__/app.test.js:169]: @nc-mark-t 
My mind went to TDD where you'd create the easiest solution first, then let the test, drive development. What do you think?

File: __tests__/app.test.js (hunk 4 above)
Comment (LukeHarrisonDev) [reply __tests__/app.test.js:179]: Thank you

File: __tests__/app.test.js (hunk 6 above)
Comment (LukeHarrisonDev) [reply __tests__/app.test.js:229]: Will sort that out. Thanks.

**Pull request merged**